arcontrol2nwb/converter_cli.py

```diff
@@ -127,88 +262,633 @@ def c_create(MAT):
+                                           cs_events: dict,
+                                           info: dict,
+                                           time_offset: float):
+    """
+    Add the data to the NWBFile using BehavioralTimeSeries and TimeIntervals
+    """
+    world_event = (io_in_events | io_out_actions | cs_events)
+    assert world_event.keys() <= info.keys()
+
+    time_len = max([v[-1, 0] + v[-1, 1] for v in world_event.values()])
 
-    nwbfile = NWBFile(
-        session_description=task_name,  # required
-        identifier=task_name+"."+filetime_str,  # required
-        session_start_time=session_start_time,  # required
-        experimenter="ArControl behavior recorder",  # optional
-    )
     behavior_module = nwbfile.create_processing_module(
         name="behavior", description="Raw ArControl event"
     )
     behavioral_events = BehavioralEvents(name="BehavioralEvents")
-    IO_event = {IO: np.array(v)/1000 for IO, v in MAT.items() 
-                    if 'IN' in IO or 'OUT' in IO}  # process IO time as sec
-    CS_pattern = re.compile('(^C\d+S\d+$)|(^C\d+$)')
-    CS_event = {CS: np.array(v)/1000 for CS, v in MAT.items()
-                    if CS_pattern.match(CS)}  # process CS time as sec
-    world_event = (IO_event|CS_event)
-    assert world_event.keys() <= MAT['info'].keys()
-
-    time_len = max([v[-1,0]+v[-1,1] for v in world_event.values()])
 
     s1 = TimeIntervals(
-            name="ArControl_Events",
-            description="intervals for each event",
-        )
+        name="ArControl_Events",
+        description="intervals for each event",
+    )
     s1.add_column(name="event", description="I/O and State events")
     for e, v in world_event.items():
         for start_t, dur_t in v:
-            s1.add_row(start_time=start_t, 
-                        stop_time=start_t+dur_t, 
-                        event=e)
+            s1.add_row(start_time=start_t + time_offset,
+                       stop_time=start_t + time_offset + dur_t,
+                       event=e)
     _ = nwbfile.add_time_intervals(s1)
 
     for e, v in world_event.items():
-        tbg_pre, tend=v[:,0], v[:,0]+v[:,1]
+        tbg_pre, tend = v[:, 0], v[:, 0] + v[:, 1]
         ddt = 0.0001
-        tend[tend==tbg_pre] += ddt
-        tbg =  tbg_pre + ddt
+        tend[tend == tbg_pre] += ddt
+        tbg = tbg_pre + ddt
         tend_post = tend + ddt
         seq_t_1 = np.concatenate((tbg, tend))
         seq_t_0 = np.concatenate((tbg_pre, tend_post))
         seq_v_1 = np.ones(seq_t_1.shape)
         seq_v_0 = np.zeros(seq_t_0.shape)
-        seq_t = np.concatenate((seq_t_0, seq_t_1, [time_len+ddt]))
+        seq_t = np.concatenate((seq_t_0, seq_t_1, [time_len + ddt]))
         seq_v = np.concatenate((seq_v_0, seq_v_1, [0]))
-        if not np.any(seq_t==0.0):
+        if not np.any(seq_t == 0.0):
             seq_t = np.append(seq_t, 0.0)
             seq_v = np.append(seq_v, 0)
```

Comment:
@chenxinfeng4 could you clarify what is code is doing? In particular, it is unclear to me:

 1) how is ``ddt`` being used here?  It looks like it used as some form of time delta to adjust timestamps. Why do timestamps need to be changed? 
2) It looks like a single event ``e`` with timerange ``v`` is here being transformed into a sequence with multiple steps. Is the purpose of this to put `0` values before and after the actual occurrence of the event which is marked by `1`?

Reply:
![image](https://user-images.githubusercontent.com/17546446/220948045-243307d5-2466-4961-b3ad-e579bc111a55.png)

One event is originally defined as [tbg, tend], or [tbg, tdur]. 
However, when converting to NWB timeseries, it should be [tbg_pre, tbg, tend, tend_post].

Reply:
I'll fix that by myself.

Reply:
Thanks for the clarification. That makes sense. For the ndx-beadl representation of the data we then should not need to do these transformation since the timings are here represented via ``start_time, end_time`` (for states) and ``timestep, duration`` (for events/actions), so that this is more closely aligned with how ARControl describes it. The main part then to look out for is that the duration for events/action is being set correctly.